darshan-util/pydarshan/tests/test_plot_io_cost.py

```diff
@@ -0,0 +1,139 @@
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            "MPI-IO",
+            pd.DataFrame(
+                np.array([[0.0196372866, 0.134251, 0.0475]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            "STDIO",
+            pd.DataFrame(
+                np.array([[0.0, 0.0001022815, 0.0]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            "POSIX",
+            pd.DataFrame(
+                np.array([[0.0, 49.022266, 0.005518]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            "MPI-IO",
+            pd.DataFrame(
+                np.array([[0.0, 49.022411, 0.023936]]),
+                ["Avg. Operation Time"],
+                ["Read", "Write", "Meta"],
+            ),
+        ),
+    ],
+)
+def test_get_io_cost_df(report, mod, expected_df):
+    # regression test for `plot_io_cost.get_io_cost_df()`
+    actual_df = get_io_cost_df(report=report, mod_key=mod)
+    assert_frame_equal(actual_df, expected_df)
+
+
+@pytest.mark.parametrize("mod", ["POSIX", "MPI-IO", "STDIO"])
+@pytest.mark.parametrize(
+    "report, expected_ylims", [
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            [0.0, 1.0],
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
+            [0.0, 120.0],
+        ),
+    ],
+)
+def test_plot_io_cost_ylims(report, expected_ylims, mod):
+    # test the y limits for both axes for the IO cost stacked bar graph
+    fig = plot_io_cost(report=report, mod_key=mod)
+    for i, ax in enumerate(fig.axes):
+        # there are only 2 axes, the first being the "raw" data
+        # and the second being the normalized data (percent)
+        actual_ylims = ax.get_ylim()
+        if i == 0:
+            assert_array_equal(actual_ylims, expected_ylims)
+        else:
+            # normalized data is always the same
+            assert_array_equal(actual_ylims, [0.0, 100.0])
+
+@pytest.mark.parametrize("mod", ["POSIX", "MPI-IO", "STDIO"])
+@pytest.mark.parametrize(
+    "report, expected_yticks, expected_yticklabels", [
+        (
+            darshan.DarshanReport("examples/example-logs/ior_hdf5_example.darshan"),
+            [0.0, 0.2, 0.4, 0.6, 0.8, 1.0],
+            ["0.0", "0.2", "0.4", "0.6", "0.8", "1.0"],
+        ),
+        (
+            darshan.DarshanReport("examples/example-logs/sample.darshan"),
```

Comment:
Here and elsewhere--I thought `sample.darshan` had buggy data in `STDIO_F_WRITE_TIME`? It may not matter, though I wonder about the long-term wisdom of leaning on it for regression tests that are added in new PRs? Or at least those that actually access that field at least.

Have issues been opened to carve a path forward on handling this log and its renamed duplicates? For example, if the Python machinery might eventually raise an exception via a lightweight sanity checker for input logs that might influence a decision to use the log (and its clones) for new testing.

Reply:
Yes, it does have a negative value that shouldn't be there. I wasn't worried about it for this particular test, but I see what you mean.

I think you raise some good points, if a sanity checker is implemented this would cause a lot of issues for the current testing suite, and reusing this log is only going to make things worse. I *thought* we had agreed to keep 1 of the logs and just remove the duplicates, but looking back at the messages it seems no hard lines have been drawn. 

I think there are a few things to discuss/issues to open from that discussion:
 1. What to do with the duplicate logs
 2. Whether we should keep `sample.darshan` at all 
 3. Where we want the logs to be stored long-term (everything in the logs repo, a mix like we currently have, etc.)
 4. What we want to do with `/examples` (and its logs) long-term
 5. Whether or not we want to add some sanity checks for logs, and how we want to do that

Reply:
Yeah, I think it would be better to avoid using this log for most testing purposes. It may still be useful to have around as an example log with bad data (in case we come up with something to sanity check log data that this could be used as input for), but other than that it may be dangerous. I'd suggest we just rename it (`bad-counters.darshan` or something like that) and remove all duplicates, and switch tests that use it over to something less problematic.

Reply:
I've purged `sample.darshan` from the tests here in lieu of `sample-badost.darshan`.